In [8]:
import pandas as pd
import matplotlib.pyplot as plt   
%matplotlib inline
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
import numpy as np
import statsmodels.formula.api as smf

In [9]:
df = pd.read_csv('salary.csv')

In [10]:
df = df.dropna(how='any',axis=0)

In [11]:
train, test = train_test_split(df, test_size=0.3)

In [12]:
X_train = train['yearsworked']
y_train = train['salary']

In [13]:
model = sm.OLS(y_train,X_train).fit()

In [14]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.752
Model:                            OLS   Adj. R-squared:                  0.751
Method:                 Least Squares   F-statistic:                     1086.
Date:                Thu, 28 Mar 2019   Prob (F-statistic):          1.81e-110
Time:                        11:05:51   Log-Likelihood:                -4161.3
No. Observations:                 359   AIC:                             8325.
Df Residuals:                     358   BIC:                             8329.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yearsworked  2808.0155     85.202     32.957      

### Does the model significantly predicts the dependent variable? Report the amount of variance explained (R^2) and significance value (p) to support your answer.

The model significantly predicts the dependent variable. This is shown from variance (R^2) 0.766(higher value generally indicates a better fit) with a p value of 0.000... (less than 0.05 is considered to be statistically significant)


### What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?
76.5%



### What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
The coeffecient is 2840.5341 this means that when salary increases by 1, the number of salary increases by 2840.5341

### What do the 95% confidence intervals [0.025, 0.975] mean?

It shows the amount you would expect at a 95% confidence level. 

### Calculate the expected salary for someone with 12 years’ work experience.

In [25]:
X_test = sm.add_constant(X_test)
model.predict(X_train[12])

array([8424.04655022])

### Calculate the expected salary for someone with 80 years’ work experience. Are there any problems with this prediction? If so, what are they?

### We have only looked at the number of years an employee has worked. What other employee characteristics might influence their salary?

### How does your model compare when running it on the test set - what is the difference in the Root Mean Square Error (RMSE) between the training and test sets?

In [19]:
X_test = test['yearsworked']
y_test= test['salary']


/home/umuzi/.local/lib/python3.6/site-packages/numpy/core/fromnumeric.py:2389: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)


In [20]:
test_model = sm.OLS(y_test,X_test).fit()

In [21]:
print_test_model = test_model.summary()
print(print_test_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.454
Model:                            OLS   Adj. R-squared:                  0.450
Method:                 Least Squares   F-statistic:                     126.2
Date:                Thu, 28 Mar 2019   Prob (F-statistic):           1.07e-21
Time:                        11:06:19   Log-Likelihood:                -1632.9
No. Observations:                 154   AIC:                             3270.
Df Residuals:                     152   BIC:                             3276.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const        3.822e+04   1339.187     28.538      